### PR TITLE
Mark book as read

### PR DESCRIPTION
**Pull request by pankaj-ag**
#364

- Give read/unread option on book info page.
- UI changes as per requirnments
- Show a Random book when the user comes on the home screen.

### CODE REVIEW

File: app/Models/KnowledgeCafe/Library/Book.php [hunk 1]
```diff
@@ -37,4 +38,21 @@ public static function getByCategoryName($categoryName) {
...
+         $result = ($read) ? $this->readers()->attach(auth()->user()) 
+                  : $this->readers()->detach(auth()->user());
+
+        return true;
+    }
+
+    public static function getRandomUnreadBook() {
+        return self::whereDoesntHave('readers', function ($query) {
+            $query->where('id', auth()->id());
+        })->get()->random();
```
Comment (bohraji) [inline app/Models/KnowledgeCafe/Library/Book.php]: Can we use `inRandomOrder` and `first` eloquent functions to reduce the overhead of fetching all the books of user.

File: resources/assets/js/app.js [hunk 2]
```diff
@@ -545,3 +544,53 @@ if (document.getElementById('books_category')) {
...
+        }
+    });
+}
+
+if(document.getElementById('home_page')) {
+    var el = document.getElementById("markBookAsRead");
+    el.addEventListener("click", markBookAsRead, false); 
+    let isModalShown = sessionStorage.getItem('book_modal_has_shown');
+    if(!isModalShown) {
+        sessionStorage.setItem("book_modal_has_shown", "true");
```
Comment (bohraji) [inline resources/assets/js/app.js]: 👍

**Comment by bohraji** [comment]
Checked this PR :

- [x] For duplication.
- [x] For performance. 
- [x] Usage of recursive methods, loops and algorithms.
- [x] For readability and understanding. 
- [x] For functions with multiple responsibilities.
- [x] For scope of refactoring. 
- [x] For coupling
- [x] For alternative approaches to the implementation. 
- [x] For how the change impacts other parts of the system.
- [x] For validations. client side and server side validations are present.

For more insights on code reviews checkout our [guidelines](https://docs.google.com/document/d/1Sa-XYJEmGwrOM5gk5zoan_Z6j90ylibmCmt331361nw/edit#).

File: app/Models/KnowledgeCafe/Library/Book.php (hunk 1 above)
Comment (pankaj-ag) [reply app/Models/KnowledgeCafe/Library/Book.php]: Great suggestion. done

**Comment by pankaj-ag** [comment]
@abhi1203 as we figured that we need to add a check on the server side also so that it will not make a random book query on every home URL request. I am tracking that on high priority in another issue #365.

**Pull request merged**